packages/components/src/styles/components/flyout.scss

```diff
@@ -17,8 +17,10 @@
 }
 
 .hds-flyout {
-  z-index: 49; // modal component is 50; we want to make sure the flyout always stays below the modal, regardless the DOM order
+  all: revert; // this must be first
```

Comment:
in reality I don't think we need to revert everything. the only thing we need to apply are the `top/right/bottom: 0` styles, no?

Reply:
after discussion with @alex-ju this morning and with @KristinLBradley this evening, we think there is a possible path forward to continue using the `<dialog>` HTML tag even if it comes with "baggage" (the user-agent CSS, the JS methods, the `::backdrop` pseudo-element, that we don't use btw)
- make sense to use the `all: unset;` declaration in the dialog primitive
- instead of using the `all: revert` in the Modal and Flyout, we can add the "missing styles" for the fixed layout, that is the `inset: 0`, the `width/height/max-height/max-width` where necessary (depends if it's a Flyout or a Modal), etc.

Once these changes are done, we can have a final look and see if we all agree to continue using the `<dialog>` for this component.